cpp/Esri/ArcGISRuntime/Toolkit/PopupViewController.h

```diff
@@ -0,0 +1,139 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_POPUPVIEWCONTROLLER_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_POPUPVIEWCONTROLLER_H
+
+// ArcGISRuntime headers
+#include <PopupManager.h>
+
+// Qt headers
+#include <QAbstractListModel>
+#include <QObject>
+
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+
+namespace Toolkit
+{
+
+/*!
+ * \brief In MVC architecture, this is the controller for the corresponding
+ * PopupView.
+ * 
+ * This controller is a thin wrapper around a PopupManager. It re-exposes some
+ * PopupManager properties, including the number of total rows to render as a 
+ * property.
+ */
+class PopupViewController : public QObject
+{
+  Q_OBJECT
+  Q_PROPERTY(PopupManager* popupManager READ popupManager WRITE setPopupManager NOTIFY popupManagerChanged)
+  Q_PROPERTY(QAbstractListModel* displayFields READ displayFields NOTIFY popupManagerChanged)
+  Q_PROPERTY(int fieldCount READ fieldCount_ NOTIFY fieldCountChanged)
+  Q_PROPERTY(QAbstractListModel* attachments READ attachments NOTIFY popupManagerChanged)
+  Q_PROPERTY(int attachmentCount READ attachmentCount_ NOTIFY attachmentCountChanged)
+  Q_PROPERTY(QString title READ title NOTIFY popupManagerChanged)
+public:
+  /*!
+   * \brief Constructor
+   * \param parent Parent owning QObject.
+   */
+  explicit Q_INVOKABLE PopupViewController(QObject* parent = nullptr);
+
+  /*!
+   * \brief Destructor.
+   */
+  ~PopupViewController();
+
+  /*!
+   * \brief Returns the PopupManager that populates this controller with data.
+   */
+  PopupManager* popupManager() const;
+
+  /*!
+   * \brief Sets the PopupManager. Setting this will trigger a notify on all
+   * remaining properties. 
+   * \param popupManager To deliver data from.
+   */
+  void setPopupManager(PopupManager* popupManager);
+
+  /*!
+   * \brief Returns a list model containing the key/value fields of the
+   * Popup associated with this PopupManager.
+   * \note This can be null.
+   */
+  QAbstractListModel* displayFields() const;
+  
+  /*!
+   * \brief Returns a list model containing the attachment images of the
+   * Popup associated with this PopupManager. 
+   * \note This can be null.
+   */
+  QAbstractListModel* attachments() const;
+
+  /*!
+   * \brief Returns the title of the PopupManager.
+   * \note This is re-exposed from PopupManager as PopupManager does not have
+   * NOTIFY/CONSTANT modifiers on its title property, so the Controller
+   * re-exposes title to suppress warnings about ths.
```

Comment:
```suggestion
   * re-exposes title to suppress warnings about this.
```

Also, I don't think we need the note here - users just need to know they can get the title here, they don't need to understand the implementation rationale IMO